### DISCUSSION TITLE
load() | callback

### DISCUSSION THREAD

**Discussion by levaweb** [katspaugh/wavesurfer.js#3203] [Q&A] [answered]
Hi, is there any callback for this method?

It works, but the next line of code won't get executed.

```
wavesurfer.load(url);
$('#modal_load').modal('hide');     // this won't execute
```

Thanks

https://github.com/katspaugh/wavesurfer.js/discussions/2876  ** loading a file

**Comment by katspaugh** [comment, marked as answer]
You can listen to the ready event, it’s fired after the audio is loaded and rendered.

Also, load returns a promise so you can “await” it.

**Reply by levaweb** [reply]
Brilliant! it worked.

I've seen that load() returns a promise, but I'm not sure how do I process it?
Thanks

**Reply by katspaugh** [reply]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
